### PR TITLE
Exposed the equation and prop_conv fields in bmct

### PR DESCRIPTION
**Pull request by dcattaruzza**

### CODE REVIEW

**Comment by tautschnig** [comment]
I don't understand the body of the commit message ("Added function that allows consumers of this class to explicitly build a goto_tracet"). Also, the subject of the commit message is longer than recommended and leaves questions open ("exposed" - to whom? what is a "simple" function?).

**Review by tautschnig — COMMENTED** [review]
Why does this particular API make sense? It seems quite arbitrary as you could instead provide read-only access functions to `equation` and `prop_conv`. I'm not saying that that would be better, but since this is just additional API without any current user it's impossible to tell whether one or the other is more appropriate.

**Comment by dcattaruzza** [comment]
Changed the tite and exposed read-only reference as per @tautschnig comments

File: src/cbmc/bmc.h [hunk 1]
```diff
@@ -71,6 +71,20 @@ class bmct:public safety_checkert
...
+  /// Retrieves the bmc equation
+  /// @return a reference to the bmc equation object
+  const symex_target_equationt& get_equation() const
+  {
+    return equation;
+  }
+
+  /// Retrieves the prop_conv field
+  /// @return a reference to the prop_conv member
+  const prop_convt & get_prop_conv() const
```
Comment (tautschnig) [inline src/cbmc/bmc.h]: Please use consistent spacing around `&` - see also clang-format's output: it should be `prop_convt &get_prop_conv`. Fix `get_equation` above accordingly.

**Review by cesaro — APPROVED** [review]

**Comment by cesaro** [comment]
@smowton Can you have a look to this? It's very simple!

**Comment by peterschrammel** [comment]
It's unclear to me what this is for. If the (suspected) goal is to build a goto-trace outside bmct (not sure why this is needed), why not expose a function that returns the goto_trace instead of exposing the internals of bmct?

**Comment by smowton** [comment]
I've really got to defer to Peter on the architectural question

**Comment by dcattaruzza** [comment]
Cancelled because it was deemed the functionality it enables should not be added.

**Pull request closed without merging**